scripts/templateCombinator.ts

```diff
@@ -0,0 +1,465 @@
+          _.createTypeReferenceNode(_.createIdentifier(name), undefined),
+        ])
+      ),
+      sinceAndCategory('Model', packageVersion),
+      _.createTypeAliasDeclaration(
+        [_.createModifier(ts.SyntaxKind.ExportKeyword)],
+        _.createIdentifier('SchemableParams1'),
+        [
+          _.createTypeParameterDeclaration(
+            undefined,
+            _.createIdentifier('S'),
+            _.createTypeReferenceNode(_.createIdentifier('URIS'), undefined),
+            undefined
+          ),
+        ],
+        _.createTypeReferenceNode(_.createIdentifier('Kind'), [
+          _.createTypeReferenceNode(_.createIdentifier('S'), undefined),
+          _.createTypeReferenceNode(_.createIdentifier(name), undefined),
+        ])
+      ),
+      sinceAndCategory('Model', packageVersion),
+      _.createTypeAliasDeclaration(
+        [_.createModifier(ts.SyntaxKind.ExportKeyword)],
+        _.createIdentifier('SchemableParams2C'),
+        [
+          _.createTypeParameterDeclaration(
+            undefined,
+            _.createIdentifier('S'),
+            _.createTypeReferenceNode(_.createIdentifier('URIS2'), undefined),
+            undefined
+          ),
+        ],
+        _.createTypeReferenceNode(_.createIdentifier('Kind2'), [
+          _.createTypeReferenceNode(_.createIdentifier('S'), undefined),
+          _.createKeywordTypeNode(ts.SyntaxKind.UnknownKeyword),
+          _.createTypeReferenceNode(_.createIdentifier(name), undefined),
+        ])
+      ),
+      sinceAndCategory('Refinements', packageVersion),
+      _.createVariableStatement(
+        [_.createModifier(ts.SyntaxKind.ExportKeyword)],
+        _.createVariableDeclarationList(
+          [
+            _.createVariableDeclaration(
+              _.createIdentifier(`is${name}`),
+              undefined,
+              undefined,
+              _.createArrowFunction(
+                undefined,
+                undefined,
+                [
+                  _.createParameterDeclaration(
+                    undefined,
+                    undefined,
+                    _.createIdentifier('s'),
+                    undefined,
+                    _.createKeywordTypeNode(ts.SyntaxKind.StringKeyword),
+                    undefined
+                  ),
+                ],
+                _.createTypePredicateNode(
+                  undefined,
+                  's',
+                  _.createTypeReferenceNode(name)
+                ),
+                _.createToken(ts.SyntaxKind.EqualsGreaterThanToken),
+                _.createBlock(
+                  [
+                    _.createThrowStatement(
+                      _.createNewExpression(_.createIdentifier('Error'), undefined, [
+                        _.createStringLiteral('not implemented'),
+                      ])
+                    ),
+                  ],
+                  true
+                )
+              )
+            ),
+          ],
+          ts.NodeFlags.Const
```

Comment:
🙃 why don't they make the variable signifier the first argument? 🤷

Reply:
That's a great question 🤔